urllib3/util/retry.py

```diff
@@ -330,7 +338,7 @@ def increment(self, method=None, url=None, response=None, error=None,
 
         elif error and self._is_read_error(error):
             # Read retry?
-            if read is False:
+            if read is False or (method is not None and not self._is_method_retryable(method)):
```

Comment:
Hrm. I don't think I think we should allow `None` here: it's not clear to me that it's every actually going to be passed. You raise a fair point about the default value though. I'm inclined to say let's just remove the check.